extract_name.py

```diff
@@ -42,6 +42,20 @@ def main():
         sys.exit(3)
 
     workflow_file = safename(info['name'])
+
+    custom_version = os.environ.get("CUSTOM_VERSION")
+    # The sanitization regexp of the custom version
+    custom_sanitize_regexp = re.compile("^\S{,15}")
+    # if the workflow run on a new release,
+    # update the info plist version to match the same version
+    if custom_version:
+        # if the custom version is `github.ref`, then sanitize it to exact tag name
+        custom_version = custom_version.replace('refs/tags/', '')
+        sanitized_custom_version = custom_sanitize_regexp.findall(custom_version)[0]
+        if sanitized_custom_version:
+            info['version'] = custom_version
+            plistlib.writePlist(info, info_plist)
+
     if 'version' in info:
         workflow_file += '-{}'.format(info['version'])
```

Comment:
Actually you don't need to extract the tag name as `github.ref_name` already contains the information that you want as opposed to `github.ref`:

<img width="727" alt="image" src="https://user-images.githubusercontent.com/43891734/210250547-a9930699-79e1-4352-a1c1-dd1452bdf625.png">

https://docs.github.com/en/actions/learn-github-actions/contexts

```suggestion

    custom_version = os.environ.get("CUSTOM_VERSION")
    if custom_version:
        info['version'] = custom_version
        plistlib.writePlist(info, info_plist)

    if 'version' in info:
        workflow_file += '-{}'.format(safename(info['version']))
```